cirq-core/cirq/circuits/qasm_output_test.py

```diff
@@ -590,3 +590,30 @@ def test_reset():
 reset q[1];
     """.strip()
     )
+
+
+def test_different_sized_registers():
+    qubits = cirq.LineQubit.range(2)
+    c = cirq.Circuit(cirq.measure(qubits[0], key='c'), cirq.measure(qubits, key='c'))
+    output = cirq.QasmOutput(
+        c.all_operations(), tuple(sorted(c.all_qubits())), header='Generated from Cirq!'
+    )
+    assert (
+        str(output).strip()
+        == """// Generated from Cirq!
+
+OPENQASM 2.0;
+include "qelib1.inc";
+
+
+// Qubits: [q(0), q(1)]
+qreg q[2];
+creg m_c[2];
+
+
+measure q[0] -> m_c[0];
+
+// Gate: cirq.MeasurementGate(2, cirq.MeasurementKey(name='c'), ())
+measure q[0] -> m_c[0];
+measure q[1] -> m_c[1];"""
+    )
```

Comment:
We can cover also the 3.0 output -
```suggestion
    )
    # OPENQASM 3.0
    output3 = cirq.QasmOutput(
        c.all_operations(),
        tuple(sorted(c.all_qubits())),
        header='Generated from Cirq!',
        version='3.0',
    )
    assert (
        str(output3)
        == """// Generated from Cirq!

OPENQASM 3.0;
include "stdgates.inc";


// Qubits: [q(0), q(1)]
qubit[2] q;
bit[2] m_c;


m_c[0] = measure q[0];

// Gate: cirq.MeasurementGate(2, cirq.MeasurementKey(name='c'), ())
m_c[0] = measure q[0];
m_c[1] = measure q[1];
"""
    )
```